common/lib/xmodule/xmodule/x_module.py

```diff
@@ -881,7 +881,22 @@ def __getattr__(self, name):
                 request_post[key] = map(FileObjForWebobFiles, request.POST.getall(key))
 
         response_data = self.handle_ajax(suffix, request_post)
-        return Response(response_data, content_type='application/json')
+
+        try:
+            return Response(response_data, content_type='application/json')
+
+        except TypeError:
+            request_environ = getattr(request, 'environ')
+            log.exception(
+                "Type Error in xmodule handler WebOb response for request url: %s, response_data type: %s, LANG: %s, "
+                "LC_ALL: %s and HTTP_ACCEPT_ENCODING: %s",
+                request_environ['HTTP_REFERER'],
```

Comment:
Why do we need `HTTP_REFERER` to log?

Reply:
To get complete problem url e.g
http://localhost:8000/courses/course-v1:edx+CS910+2017_T1/courseware/unit4/app_2/?activate_block_id=block-v1%3Aedx%2BCS910%2B2017_T1%2Btype%40sequential%2Bblock%40app_2